src/components/CollaboratorModal/submodals/AcknowledgementSubmodal.tsx

```diff
@@ -84,25 +83,17 @@ const AcknowledgementSubmodalContent = () => {
         </Text>
       </Text>
       <br />
-      <Checkbox onChange={() => setIsAcknowledged(!isAcknowledged)}>
+      <Checkbox {...register("isAcknowledged")}>
         <Text color="text.body" fontSize="16px">
           <Text as="span">I agree to Isomer&lsquo;s</Text>{" "}
           <Link href={TERMS_OF_USE_LINK}>Terms of Use</Link>
         </Text>
       </Checkbox>
       <Stack spacing={4} direction="row" justify="right">
-        <Button
-          variant="clear"
-          color="secondary"
-          onClick={() => {
-            setModalState(CollaboratorModalState.Default)
-            setIsAcknowledged(false)
-            closeModal()
-          }}
-        >
+        <Button variant="clear" color="secondary" onClick={onClose}>
           Cancel
         </Button>
-        <Button isDisabled={!isAcknowledged} onClick={handleAddCollaborator}>
+        <Button isDisabled={!isAcknowledged} type="submit">
```

Comment:
When the user clicks submit, the success toast appears but the acknowledgement modal is not closed. I think ideally the user should be brought back to the main submodal with the new collaborator shown in the collaborators list.